timedctl/timedctl.py

```diff
@@ -25,708 +24,568 @@
+            customer_id = self.get_customer_by_name(customers, customer_name, archived)
+        projects = self.timed.projects.get(
+            cached=True, filters={"customer": customer_id},
         )
-        comment = report["attributes"]["comment"]
-        duration = report["attributes"]["duration"]
-
-        table.append([customer, project, task, comment, duration])
-    # create the output
-    output = terminaltables.SingleTable(table)
-    msg(f"Reports for {date if date is not None else 'today'}:")
-    click.echo(output.table)
-    msg(f"Total: {time_sum(table)}")
-
-
-@get.command("activities", aliases=["a", "ac", "activity"])
-@click.option("--date", default=None)
-def get_activities(date):
-    """Get activities."""
-    activities = timed.activities.get(
-        filters={"day": date},
-        include="task,task.project,task.project.customer",
-    )
-    table = [["Activity", "Comment", "Start", "End"]]
-    for activity_obj in activities:
-        table.append(
-            [
-                format_activity(activity_obj),
-                activity_obj["attributes"]["comment"],
-                activity_obj["attributes"]["from-time"].strftime("%H:%M:%S"),
-                activity_obj["attributes"]["to-time"].strftime("%H:%M:%S")
-                if activity_obj["attributes"]["to-time"] is not None
-                else "active",
-            ],
+        output = []
+        for project in projects:
+            output.append({"id": project["id"], "name": project["attributes"]["name"]})
+        output_formatted(output, output_format)
+
+    def get_tasks(
+        self,
+        output_format,
+        customer_id,
+        customer_name,
+        project_id,
+        project_name,
+        archived,
+    ):
+        """Get tasks."""
+        if project_name and not (customer_id or customer_name):
+            error_handler("ERR_CUSTOMER_INFO_MISSING")
+        if not (project_id or project_name):
+            error_handler("ERR_MISSING_ARGUMENTS")
+        # Get project ID if name is specified
+        if not project_id:
+            # we need an id for the customer
+            if not customer_id:
+                customers = self.timed.customers.get(cached=True)
+                customer_id = self.get_customer_by_name(
+                    customers, customer_name, archived,
+                )
+            # get the project id
+            projects = self.timed.projects.get(
+                cached=True,
+                filters={"customer": customer_id},
+            )
+            project_id = self.get_project_by_name(
+                projects,
+                project_name,
+                customer_id,
+                archived,
+            )
+        # get the tasks for the specified project
+        tasks = self.timed.tasks.get(cached=True, filters={"project": project_id})
+        output = []
+        for task in tasks:
+            output.append({"id": task["id"], "name": task["attributes"]["name"]})
```

Comment:
Again, use a list comprehension, as this seems to be a common pattern in the app consider creating a helper method for exactly this

Reply:
will do at some point, for now I've just added list comprehensions